src/screens/DomainScreen/elements/Header/elements/ActionButtonGroup.js

```diff
@@ -1,148 +1,137 @@
-  const __renderFollowButton = () => {
-    if (follow)
-      return (
-        <GlobalButton
-          buttonStyle={styles.globalButtonFollowContainer}
-          onPress={() => handleUnfollow()}>
-          <View style={styles.buttonFollowing}>
-            <Text style={styles.textButtonFollowing}>Following</Text>
-          </View>
+const ActionButtonGroup = ({ follow, handleFollow, handleUnfollow, isBlocked, onPressBlock, onPressUnblock }) => {
+    const __renderFollowButton = () => {
+        if (follow) return <GlobalButton buttonStyle={styles.globalButtonFollowContainer}
+            onPress={() => handleUnfollow()}>
+            <View style={styles.buttonFollowing}>
+                <Text style={styles.textButtonFollowing}>Following</Text>
+            </View>
         </GlobalButton>
-      );
 
-    return (
-      <GlobalButton buttonStyle={styles.globalButtonFollowContainer} onPress={() => handleFollow()}>
-        <View style={styles.buttonFollow}>
-          <Text style={styles.textButtonFollow}>Follow</Text>
-        </View>
-      </GlobalButton>
-    );
-  };
+        return <GlobalButton buttonStyle={styles.globalButtonFollowContainer}
+            onPress={() => handleFollow()}>
+            <View style={styles.buttonFollow}>
+                <Text style={styles.textButtonFollow}>Follow</Text>
+            </View>
+        </GlobalButton>
+    }
+
+    const __renderBlockButton = () => {
+        if (!isBlocked) return <GlobalButton buttonStyle={styles.globalButtonBlockContainer}
+            onPress={() => onPressBlock()}>
+            <View style={styles.buttonBlock}>
+                <Text style={styles.blockButtonText}>Block</Text>
 
-  const __renderBlockButton = () => {
-    if (!isBlocked)
-      return (
-        <GlobalButton
-          buttonStyle={styles.globalButtonBlockContainer}
-          onPress={() => onPressBlock()}>
-          <View style={styles.buttonBlock}>
-            <Text style={styles.blockButtonText}>Block</Text>
-          </View>
+            </View>
         </GlobalButton>
-      );
 
-    return (
-      <GlobalButton
-        buttonStyle={styles.globalButtonBlockContainer}
-        onPress={() => onPressUnblock()}>
-        <View style={styles.buttonUnblock}>
-          <Text style={styles.unblockButtonText}>Blocked</Text>
-        </View>
-      </GlobalButton>
-    );
-  };
+        return <GlobalButton buttonStyle={styles.globalButtonBlockContainer}
+            onPress={() => onPressUnblock()}>
+            <View
+                style={styles.buttonUnblock}
+            >
+                <Text style={styles.unblockButtonText}>Blocked</Text>
+            </View>
+        </GlobalButton>
+    }
 
-  return (
-    <View style={styles.container}>
-      {__renderFollowButton()}
-      {__renderBlockButton()}
+    return <View style={styles.container}>
+        {__renderFollowButton()}
+        {__renderBlockButton()}
     </View>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in this section seems fine. The component renders two buttons: Follow/Unfollow and Block/Unblock based on the state of `follow` and `isBlocked`. It's important to ensure that the functions `handleFollow`, `handleUnfollow`, `onPressBlock`, and `onPressUnblock` are properly defined and handle all edge cases.



<!-- This is an auto-generated comment by OSS CodeRabbit -->